index.d.ts

```diff
@@ -795,20 +795,20 @@ declare namespace nkruntime {
     /**
      * Match handler definitions
      */
-    export interface MatchHandler {
-        matchInit: MatchInitFunction;
-        matchJoinAttempt: MatchJoinAttemptFunction;
-        matchJoin: MatchJoinFunction;
-        matchLeave: MatchLeaveFunction;
-        matchLoop: MatchLoopFunction;
-        matchTerminate: MatchTerminateFunction;
-        matchSignal: MatchSignalFunction;
+    export interface MatchHandler<State> {
+        matchInit: MatchInitFunction<State>;
+        matchJoinAttempt: MatchJoinAttemptFunction<State>;
+        matchJoin: MatchJoinFunction<State>;
+        matchLeave: MatchLeaveFunction<State>;
+        matchLoop: MatchLoopFunction<State>;
+        matchTerminate: MatchTerminateFunction<State>;
+        matchSignal: MatchSignalFunction<State>;
     }
 
     /**
      * Match initialization function definition.
      */
-    export interface MatchInitFunction {
+    export interface MatchInitFunction<State> {
```

Comment:
Can we add generic default here as well to improve backwards compatibility please?
```suggestion
    export interface MatchInitFunction<State = MatchState> {
```
Same for the following MatchHandler function definitions